doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+processing without any external instruments to be applied by user.
+So, that's why Streaming-API is needed.
+
+The usage of this G-API feature makes codewriting easier, or, at least, more
+independent. As usual in this tutorial, let's deal with it by several steps:
+
+### 1) Pipeline graph definition {#gapi_fb_s_1}
+
+To begin with, a `cv::GComputation` object (which contains all the
+pipeline) used to be defined statically by graph's input and output described in
+a code (almost all the graph's body is skipped in this code snippet except the
+input and output nodes):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp comp_old_4
+
+To apply Streaming-API we should use a different version of a
+pipeline expression  with a lambda-based constructor to keep all temporary data
+objects in a dedicated scope:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp comp_str_4
+
+### 2) Frames processing {#gapi_fb_s_2}
+
+The next step used to be a videostream definition and a source setup:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp header
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp in_out_1
+
+and then a compiled graph used to be applied to every single frame.
+This is an example of a processing loop without Streaming:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp in_out_2
+
+Now this can be done without explicit use of additional libraries. First of all,
+define a stream variable and pass to it:
+    - video properties (this unconvenience is temporary);
+    - graph compile arguments (which will be passed to the graph inside
+the stream):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_comp
+
+Then we should set a source:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_src
+
+And for this moment our stream is ready to be launched! An example of the
+Streaming loop:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_loop
+
+There are some other useful possibilities provided, e.g. you can abort stream
+processing via `cv::GStreamingCompiled::stop()` method (the next call of
+`cv::GStreamingCompiled::start()` will start processing the stream from the very
+beginning).
+
+### 3) ??? {#gapi_fb_s_3}
```

Comment:
Don't forget to fill this

Reply:
It is filled.. ;)
The next reference is the one you've liked.. Are there too much references in a small area? Ok, maybe